From a4a9506422ebb141f359ba9f31df356f262ed763 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Tue, 10 Sep 2024 10:06:57 +0300
Subject: [PATCH] [SW-0] common: print name of the loading component

The ppboot_cp_preboot function is used to copy
preboot image, sram mbist image etc. This patch
prints the name of the image being copied from
the offset to loadaddr.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Idaace396bca564fc84b855e4be1631a48d124a19
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index a163c3e..5443b93 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -398,7 +398,9 @@
 	enum dma_error dma_err;
 	enum dma_dest_type dma_type;
 
-	preboot_puts_hex64("Preboot loadaddr: ", loadaddr, false);
+	if (name)
+		preboot_puts_raw(name);
+	preboot_puts_hex64(" loadaddr: ", loadaddr, false);
 
 	dma_type = preboot_get_dma_dest_type(loadaddr);
 
diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index aeff2ec..56b6848 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -303,7 +303,9 @@
 
 void ppboot_cp_preboot(u64 loadaddr, u32 offset, u32 size, char *name)
 {
-	preboot_puts_hex_raw("Preboot loadaddr: ",
+	if (name)
+		preboot_puts_raw(name);
+	preboot_puts_hex_raw(" loadaddr: ",
 			     (u32)(loadaddr >> 32), false);
 	preboot_puth((u32)loadaddr, false);
 
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 80a6b77..3cf073a 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -216,7 +216,9 @@
 	enum dma_dest_type dma_type;
 	bool is_psoc_mem;
 
-	preboot_puts_hex64("Preboot loadaddr: ", loadaddr, false);
+	if (name)
+		preboot_puts_raw(name);
+	preboot_puts_hex64(" loadaddr: ", loadaddr, false);
 
 	dma_dest_addr = gaudi3_dma_addr_fixup(loadaddr);
 	dma_type = preboot_get_dma_dest_type(dma_dest_addr);
